File: MainGame/game/Global/Common/BackEndUtil.cs
using System;
using Godot;

namespace ZeromaXPlayground.game.Global.Common;

public static class BackEndUtil
{
    public static Tuple<int, int> To(Vector2I vec) => new(vec.X, vec.Y);

    public static Vector2I FromI(Tuple<int, int> coord) => new(coord.Item1, coord.Item2);

    public static Vector2 From(Tuple<int, int> coord) => new(coord.Item1, coord.Item2);
}

File: MainGame/FrontEnd4IdleStrategyFS/Global/Common/BackEndUtil.fs
namespace FrontEnd4IdleStrategyFS.Global.Common

open Godot

module BackEndUtil =
    let toI (vec: Vector2I) = vec.X, vec.Y
    let fromI (x, y) = Vector2I(x, y)
    let from (x, y) = Vector2(float32 x, float32 y)
    let to3 (vec: Vector3) = vec.X, vec.Y, vec.Z
